### ISSUE TITLE
C++ modules that use C sockets don't work

### ISSUE THREAD

**Issue by bmeck** [nodejitsu/haibu-carapace#5]
zeromq and other addons that do not allow for communication over existing sockets do not work properly with the rebinding shim. The proper solution would be forcing carapaced packages to be recompiled with fake headers and providing a secondary rebinding shim in C++.

The recommended workaround is to avoid these modules and note to them this tight coupling in hopes they eventually patch their modules (or libraries that are used).

**Comment by jamesonjlee** [comment]
binding of a binding is a binding...
+1 to telling people that 'these plugins will break'

**Comment by dominictarr** [comment]
maybe what carapace is doing here is just 'too clever'.

maybe it would be simpler overall do this shimlessly with a simple convention instead. 

something like

```
module.exports = http.createServer(function () {...})

if(!module.parent)
  module.exports.listen(PORT)
```

there are just so many edge cases. 

just putting it out there, we are doing this the hard way.

**Comment by bmeck** [comment]
Yep, we are doing it the hard way, but doing it the hard way allows
for easy deployment outside of our environment as well as multiple
port binding in the future. Having a single PORT variable or external
service that needs to be called would take away this. Either way even
if we do limit a program to a single port it would not actually solve
this particular problem: the C++ addon still needs to know what it
_can_ bind to.

**Comment by dominictarr** [comment]
okay, good. just making sure it was out there, so I put it out there.
